base/range.jl

```diff
@@ -1143,6 +1143,13 @@ function intersect(r::StepRange, s::StepRange)
     step(r) < zero(step(r)) ? StepRange{T,S}(n, -a, m) : StepRange{T,S}(m, a, n)
 end
 
+function intersect(r1::AbstractRange, r2::AbstractRange)
+    if length(r2) < length(r1)
+        r1, r2 = r2, r1
+    end
+    return [x for x in r1 if x ∈ r2]
```

Comment:
There are range types like `StepRangeLen` which can have step zero and contain more than one element. So we still need some way of ensuring every element in the returned array is unique.

Reply:
Is it enough to just do 
```julia
return unique!([x for x in r1 if x ∈ r2])
```
?

Reply:
In that case, you can probably just use the approach you used for the other cases already, which should allocate less.

Reply:
I changed the test to check the uniqueness too.

Reply:
Calling `unique` on the ranges might be another way, `unique(StepRangeLen(1,0,10))` seems to do the right thing.